model/project_parser.go

```diff
@@ -644,40 +697,64 @@ func LoadProjectInto(ctx context.Context, data []byte, opts *GetProjectOpts, ide
+	if len(intermediateProject.Include) > 0 {
 		if opts == nil {
 			err = errors.New("trying to open include files with empty options")
 			return nil, errors.Wrapf(err, LoadProjectError)
 		}
-		opts.UpdateForFile(path.FileName)
-
-		var yaml []byte
-		opts.Identifier = identifier
-		opts.RemotePath = path.FileName
-		grip.Debug(message.Fields{
-			"message":     "retrieving included YAML file",
-			"remote_path": opts.RemotePath,
-			"read_from":   opts.ReadFileFrom,
-			"module":      path.Module,
-		})
-		if path.Module != "" {
-			yaml, err = retrieveFileForModule(ctx, *opts, intermediateProject.Modules, path.Module)
-		} else {
-			yaml, err = retrieveFile(ctx, *opts)
+		wg := sync.WaitGroup{}
+		yamlChan := make(chan yamlTuple, len(intermediateProject.Include))
+		includeChan := make(chan Include)
+
+		// Be polite. Don't make more than 10 concurrent requests to GitHub.
+		for i := 0; i < 10; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				for include := range includeChan {
+					processIntermediateProjectInludes(ctx, identifier, intermediateProject, include, yamlChan, opts)
+				}
+			}()
 		}
-		if err != nil {
-			return intermediateProject, errors.Wrapf(err, "%s: retrieving file '%s'", LoadProjectError, path.FileName)
+
+		for _, path := range intermediateProject.Include {
+			includeChan <- path
 		}
-		add, err := createIntermediateProject(yaml, opts.UnmarshalStrict)
-		if err != nil {
-			return intermediateProject, errors.Wrapf(err, "%s: loading file '%s'", LoadProjectError, path.FileName)
+
+		close(includeChan)
+		wg.Wait()
+		close(yamlChan)
+
+		// We must apply the includes in order, so we indirect through a map.
+		yamlMap := map[string][]byte{}
+		catcher := grip.NewBasicCatcher()
+		for elem := range yamlChan {
+			catcher.Add(elem.err)
+			// defensive programming, since there should always be an error in
+			// the nil case, causing us to return early
+			if elem.yaml != nil {
+				yamlMap[elem.name] = elem.yaml
+			}
 		}
-		err = intermediateProject.mergeMultipleParserProjects(add)
-		if err != nil {
-			return intermediateProject, errors.Wrapf(err, "%s: merging file '%s'", LoadProjectError, path.FileName)
+
+		if catcher.HasErrors() {
+			return intermediateProject, errors.Wrap(catcher.Resolve(), "getting includes")
 		}
+
+		for _, path := range intermediateProject.Include {
+			add, err := createIntermediateProject(yamlMap[path.FileName], opts.UnmarshalStrict)
+			if err != nil {
+				// Return intermediateProject even if we run into issues to show merge progress.
+				return intermediateProject, errors.Wrapf(err, "%s: loading file '%s'", LoadProjectError, path.FileName)
+			}
+			if err = intermediateProject.mergeMultipleParserProjects(add); err != nil {
+				// Return intermediateProject even if we run into issues to show merge progress.
+				return intermediateProject, errors.Wrapf(err, "%s: merging file '%s'", LoadProjectError, path.FileName)
+			}
+		}
+
```

Comment:
mega nit: remove random whitespace

Reply:
love this self awareness

Reply:
Unfortunately, it's not smart enough to make style decisions when it comes to single newlines between statements/closing brackets, so I just comment on them in the PR if they look like stray whitespace. 🤷